File: src/main/java/com/fifo/ticketing/domain/performance/dto/PerformanceDetailResponse.java
package com.fifo.ticketing.domain.performance.dto;

import java.time.LocalDateTime;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
@AllArgsConstructor
public class PerformanceDetailResponse {

    private Long performanceId;
    private String title;
    private String description;
    private String placeName;
    private String encodedFileName;
    private String address;
    private String category;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private LocalDateTime reservationStartTime;
    private boolean performanceStatus;
    private int totalSeats;
    private List<PerformanceSeatGradeDto> seatGrades;
    private String urlPrefix;

    public String getUrl() {
        return urlPrefix + encodedFileName;
    }
}

File: src/main/java/com/fifo/ticketing/domain/performance/dto/PerformanceDetailResponse.kt
package com.fifo.ticketing.domain.performance.dto

import java.time.LocalDateTime

data class PerformanceDetailResponse(
    val performanceId: Long,
    val title: String,
    val description: String,
    val placeName: String,
    val encodedFileName: String,
    val address: String,
    val category: String,
    val startTime: LocalDateTime,
    val endTime: LocalDateTime,
    val reservationStartTime: LocalDateTime,
    val performanceStatus: Boolean,
    val totalSeats: Int,
    val seatGrades: List<PerformanceSeatGradeDto>,
    val urlPrefix: String,
) {
    val url: String
        get() = urlPrefix + encodedFileName
}
